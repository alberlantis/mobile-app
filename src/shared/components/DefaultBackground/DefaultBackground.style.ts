import { StyleSheet } from "react-native";

import colors from "src/theme/colors";

export const background = StyleSheet.compose(StyleSheet.absoluteFill, {
  backgroundColor: colors.BLACK,
});
export const svgBackground = StyleSheet.compose(StyleSheet.absoluteFill, {
  opacity: 0.1,
});
