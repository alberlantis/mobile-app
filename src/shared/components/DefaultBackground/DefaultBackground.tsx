import React, { Fragment } from "react";
import {
  useWindowDimensions,
  View,
  SafeAreaView,
  StyleProp,
  ViewStyle,
} from "react-native";
import Svg, { Defs, RadialGradient, Stop, Rect } from "react-native-svg";

import colors from "src/theme/colors";
import { background, svgBackground } from "./DefaultBackground.style";

export type BlurPosition = "medium" | "bottom" | "top";
interface IDefaultBackgroundProps {
  children: React.ReactNode;
  style?: StyleProp<ViewStyle>;
  blurPos?: BlurPosition;
}

const getBlurPosition = (position: BlurPosition) => {
  switch (position) {
    case "bottom":
      return "33%";
    case "top":
      return "-33%";
    case "medium":
    default:
      return "0%";
  }
};

const DefaultBackground: React.FC<IDefaultBackgroundProps> = ({
  children,
  style,
  blurPos = "medium",
}) => {
  const { width, height } = useWindowDimensions();
  const radialGradialID = "radGrad";
  const position = getBlurPosition(blurPos);

  return (
    <Fragment>
      <View style={background} />
      <Svg width={width} height={height} style={svgBackground}>
        <Defs>
          <RadialGradient
            id={radialGradialID}
            cx="50%"
            cy="50%"
            r="50%"
            fx="50%"
            fy="50%"
          >
            <Stop offset="0%" stopColor={colors.WHITE} />
            <Stop offset="100%" stopColor={colors.BLACK} />
          </RadialGradient>
        </Defs>
        <Rect
          x="0%"
          y={position}
          width={width}
          height={height}
          fill={`url(#${radialGradialID})`}
        />
      </Svg>
      <SafeAreaView style={style}>{children}</SafeAreaView>
    </Fragment>
  );
};

export default DefaultBackground;
