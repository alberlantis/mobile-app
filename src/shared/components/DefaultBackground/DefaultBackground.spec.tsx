import { render, screen } from "@testing-library/react-native";
import { Text } from "react-native";

import DefaultBackground, { BlurPosition } from "./DefaultBackground";

const renderComponent = (position?: BlurPosition) => {
  render(
    <DefaultBackground blurPos={position} style={{ width: 80, height: 80 }}>
      <Text>Background children</Text>
    </DefaultBackground>,
  );
};

describe("DefaultBackground", () => {
  it("should render children with blur position at center on the screen", () => {
    renderComponent();
    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.getByText("Background children")).toBeOnTheScreen();
  });

  it("should render children with blur position at top on the screen", () => {
    renderComponent("top");
    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.getByText("Background children")).toBeOnTheScreen();
  });

  it("should render children with blur position at bottom on the screen", () => {
    renderComponent("bottom");
    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.getByText("Background children")).toBeOnTheScreen();
  });
});
