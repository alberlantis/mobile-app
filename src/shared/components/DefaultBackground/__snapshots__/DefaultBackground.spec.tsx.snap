// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultBackground should render children with blur position at bottom on the screen 1`] = `
[
  <View
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "backgroundColor": "#000",
        },
      ]
    }
  />,
  <RNSVGSvgView
    bbHeight={1334}
    bbWidth={750}
    focusable={false}
    height={1334}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "opacity": 0.1,
          },
        ],
        {
          "flex": 0,
          "height": 1334,
          "width": 750,
        },
      ]
    }
    width={750}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGDefs>
        <RNSVGRadialGradient
          cx="50%"
          cy="50%"
          fx="50%"
          fy="50%"
          gradient={
            [
              0,
              -1,
              1,
              -16777216,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="radGrad"
          rx="50%"
          ry="50%"
        />
      </RNSVGDefs>
      <RNSVGRect
        fill={
          {
            "brushRef": "radGrad",
            "type": 1,
          }
        }
        height={1334}
        propList={
          [
            "fill",
          ]
        }
        width={750}
        x="0%"
        y="33%"
      />
    </RNSVGGroup>
  </RNSVGSvgView>,
  <View
    onLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "#000",
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
      scrollEnabled={false}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <RCTSafeAreaView
          style={
            {
              "height": 80,
              "width": 80,
            }
          }
        >
          <Text>
            Background children
          </Text>
        </RCTSafeAreaView>
      </View>
    </RCTScrollView>
  </View>,
]
`;

exports[`DefaultBackground should render children with blur position at center on the screen 1`] = `
[
  <View
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "backgroundColor": "#000",
        },
      ]
    }
  />,
  <RNSVGSvgView
    bbHeight={1334}
    bbWidth={750}
    focusable={false}
    height={1334}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "opacity": 0.1,
          },
        ],
        {
          "flex": 0,
          "height": 1334,
          "width": 750,
        },
      ]
    }
    width={750}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGDefs>
        <RNSVGRadialGradient
          cx="50%"
          cy="50%"
          fx="50%"
          fy="50%"
          gradient={
            [
              0,
              -1,
              1,
              -16777216,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="radGrad"
          rx="50%"
          ry="50%"
        />
      </RNSVGDefs>
      <RNSVGRect
        fill={
          {
            "brushRef": "radGrad",
            "type": 1,
          }
        }
        height={1334}
        propList={
          [
            "fill",
          ]
        }
        width={750}
        x="0%"
        y="0%"
      />
    </RNSVGGroup>
  </RNSVGSvgView>,
  <View
    onLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "#000",
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
      scrollEnabled={false}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <RCTSafeAreaView
          style={
            {
              "height": 80,
              "width": 80,
            }
          }
        >
          <Text>
            Background children
          </Text>
        </RCTSafeAreaView>
      </View>
    </RCTScrollView>
  </View>,
]
`;

exports[`DefaultBackground should render children with blur position at top on the screen 1`] = `
[
  <View
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "backgroundColor": "#000",
        },
      ]
    }
  />,
  <RNSVGSvgView
    bbHeight={1334}
    bbWidth={750}
    focusable={false}
    height={1334}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "opacity": 0.1,
          },
        ],
        {
          "flex": 0,
          "height": 1334,
          "width": 750,
        },
      ]
    }
    width={750}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGDefs>
        <RNSVGRadialGradient
          cx="50%"
          cy="50%"
          fx="50%"
          fy="50%"
          gradient={
            [
              0,
              -1,
              1,
              -16777216,
            ]
          }
          gradientTransform={null}
          gradientUnits={0}
          name="radGrad"
          rx="50%"
          ry="50%"
        />
      </RNSVGDefs>
      <RNSVGRect
        fill={
          {
            "brushRef": "radGrad",
            "type": 1,
          }
        }
        height={1334}
        propList={
          [
            "fill",
          ]
        }
        width={750}
        x="0%"
        y="-33%"
      />
    </RNSVGGroup>
  </RNSVGSvgView>,
  <View
    onLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "#000",
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
      scrollEnabled={false}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <RCTSafeAreaView
          style={
            {
              "height": 80,
              "width": 80,
            }
          }
        >
          <Text>
            Background children
          </Text>
        </RCTSafeAreaView>
      </View>
    </RCTScrollView>
  </View>,
]
`;
